paasta_tools/flink_tools.py

```diff
@@ -166,6 +186,56 @@ def _dashboard_get(cr_name: str, cluster: str, path: str) -> str:
     return response.text
 
 
+def _filter_for_endpoint(json_response: Any, endpoint: str) -> Mapping[str, Any]:
+    """
+    Filter json response to include only a subset of fields.
+    """
+    if endpoint == "config":
+        return {
+            key: value for (key, value) in json_response.items() if key in CONFIG_KEYS
+        }
+    if endpoint == "overview":
+        return {
+            key: value for (key, value) in json_response.items() if key in OVERVIEW_KEYS
+        }
+    if endpoint == "jobs":
+        return json_response
+    if endpoint.startswith("jobs"):
+        return {
+            key: value
+            for (key, value) in json_response.items()
+            if key in JOB_DETAILS_KEYS
+        }
+    return json_response
+
+
+def _get_dashboard_url_from_flink_cr(cr: Mapping[str, Any]) -> str:
+    return cr["metadata"]["annotations"]["flink.yelp.com/dashboard_url"]
+
+
+def curl_flink_endpoint(cr_id: Mapping[str, str], endpoint: str) -> Mapping[str, Any]:
+    try:
+        cr = get_cr(settings.kubernetes_client, cr_id)
+        if cr is None:
+            raise ValueError(f"failed to get CR for id: {cr_id}")
+        base_url = _get_dashboard_url_from_flink_cr(cr)
```

Comment:
interesting naming - wouldn't expect a dashboard to serve api requests :p

Reply:
What can I say... I agree it's not great :) The problem is the base url of the rest api is actually the same as the full url of the flink dashboard, and we're storing that info in the CR as an annotation called `flink.yelp.com/dashboard_url`. I'll change the name of the func and add a comment, but pls let me know if you have any other suggestion :)